### PR TITLE
Feature/user solutions

### LINKED ISSUE #12: Make challenge order changeable

### LINKED ISSUE #24: Hide later challenges from the user
The user should be able to access only the next uncompleted challenge, and the challenges they have completed. This feature requires keeping track which challenges which users has done (related to #10)

### PR DESCRIPTION
**Pull request by joeljuhela**
Fixes #12
Fixes #24

### CODE REVIEW

**Comment by felixbade** [comment]
Next challenge button on the front page is broken

**Comment by joeljuhela** [comment]
I think I know what causes this. It's because by default there is no Challenge in the database for first-challenge and second-challenge. 

This shouldn't be an issue when challenges are created through the platform

**Comment by joeljuhela** [comment]
Nevertheless, it should be fixed

**Comment by felixbade** [comment]
I think that solving a challenge again should navigate the same way as when solving the challenge for the first time. E.g. it's not an error, but goes forward. Just doesn't add anything to the scoreboard

**Comment by joeljuhela** [comment]
> Next challenge button on the front page is broken

A current workaround for the example challenges is to delete them and recreate the through the admin panel

**Comment by felixbade** [comment]
ah yeah I only had the challenges in the filesystem. 😄 I think this is not a problem that needs to be fixed as it will not affect any future instances, only the one on my own computer (because the challenges were created with a previous version of the platform)

**Comment by felixbade** [comment]
however, could make sense to move `get_challenge_list` in [`app/controllers/challenge.py`](https://github.com/felixbade/ctf-platform/pull/35/files#diff-a6ff994dec864dc296570dd99fe35c04a8dfdaa739574aac61d1df7ec06a9899) to use the db version of challenge enumeration as well, so the admin panel view would be consistent with what players experience

**Comment by felixbade** [comment]
basically getting rid of `order.txt`

File: app/controllers/challenge.py [hunk 1]
```diff
@@ -1,35 +1,87 @@
...
+            ).first()
+        else:
+            challenge_num = latest_user_solution.challenge.order_num
+            challenge_accessed = Challenge.query.filter(
+                (Challenge.order_num <= challenge_num + 1) & (Challenge.title == challenge_title)
+            ).first()
+
+        if not challenge_accessed:
+            # This will be returned if the queries did not return any Challenges
+            return login_manager.unauthorized()
```
Comment (felixbade) [inline app/controllers/challenge.py:44]: I think this should be abort(403). Now it gives 401, which means the user needs to log in.

Comment (felixbade) [reply app/controllers/challenge.py:44]: Made an issue (#45)

**Pull request merged**